python/smqtk/utils/bit_utils.py

```diff
@@ -143,32 +194,16 @@ def int_to_bit_vector(integer, bits=0):
     :param integer: integer to convert
     :type integer: int
 
-    :param bits: Optional fixed number of bits that should be represented by the
-        vector.
+    :param bits: Optional fixed number of bits that should be represented by
+        the vector.
     :type bits: Optional specification of the size of returned vector.
 
     :return: Bit vector as numpy array (big endian), or None if too few
         ``bits`` were specified to contain the result.
     :rtype: numpy.ndarray[bool]
 
     """
-    # Converting integer to array
-    if integer:
-        # Can-t use math version because floating-point precision runs out after
-        # about 2^48
-        # -2 to remove length of '0b' string prefix
-        size = len(bin(integer)) - 2
-    else:
-        size = 0
-    if bits and (bits - size) < 0:
-        return None
-
-    v = numpy.zeros(bits or size, numpy.bool_)
-    for i in xrange(0, size):
-        v[-(i+1)] = integer & 1
-        integer >>= 1
-
-    return v
+    return _int_to_bit_vector_fast(integer, bits)
```

Comment:
Same as above, can numba jit compile this?

Reply:
No, you're absolutely right. numba actually imposes a performance penalty here. I also redid the benchmarking for this, and _int_to_bit_vector_fast still outperforms the numba-compiled original version, even with compilation time properly handled.

Reply:
Thanks very much for the feedback so far!

I did test the knn variants and didn't see a substantial performance difference among them. vp_knn_iterative was *slightly* faster than the recursive variant for vp trees, but vps_knn_recursive was *slightly* faster for vps trees. That being said, those tests were for 1000-10000 stored hashes, so I'll retest them at the scales you mentioned.

I'll also be happy to run those plots for up to 100 million entries. I'll post them here as soon as I've got them.